test/unit/binshard.c

```diff
@@ -0,0 +1,101 @@
+#include "test/jemalloc_test.h"
+
+const char *malloc_conf = "narenas:1,bin_shards:1-128:16|129-512:4";
```

Comment:
I think for Windows, this needs to be set in a wrapper script, the way we do with other malloc_conf tests.

Let's test a couple weirder settings, too? (e.g. too many bins for us to represent, a bin with 0, individual size classes picked out from larger ranges for a different setting, etc.).

Reply:
re: the weird settings, the current conf_init errors out on the invalid settings (in `bin_update_shard_size()`), which should be better than silently dropping the option. Combined with abort_conf enabled in dbg mode, it will crash in these cases. I did verify it locally though. Do you prefer to add "abort_conf:false" explicitly for the test case? It will still output the error msg which could be misleading as well.

Reply:
re: the wrapper -- yep will change it to there.

Reply:
Added test case for overlapping ranges and a individual value from a previous larger range.

Reply:
TBH I've never been clear on how we feel about contact-violating malloc_conf options. One the one hand, we try to do something reasonable in the case of nonsensical configurations, which implies that maybe we want to be resilient to them. But on the other, we don't really test or validate the parsing code the way we would if it had to deal with genuinely hostile inputs.